app/Http/Controllers/UniversitiesController.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Validator;
+use GuzzleHttp\Client;
+use GuzzleHttp\Promise;
+
+class UniversitiesController extends Controller
+{
+    /**
+     * Create a new JSON response instance.
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+
+    public function searchUniversities(Request $request)
+	{
+	    $query = $request->input('q');
+	    $newArray = [];
+	    if($query){
+			$client = new Client();
+			
+			$limit = 5;
+			$promises = [
+			    'byCountry' => $client->getAsync('http://universities.hipolabs.com/search', [
+			        'query' => [
+			            'country' => $query,
+			            'limit'=>$limit
+			        ],
+			    ]),
+			    'byName' => $client->getAsync('http://universities.hipolabs.com/search', [
+			        'query' => [
+			            'name' => $query,
+			            'limit'=>$limit
+			        ],
+			    ]),
+			    'byDomain' => $client->getAsync('http://universities.hipolabs.com/search', [
+			        'query' => [
+			            'domain' => $query,
+			            'limit'=>$limit
+			        ],
+			    ]),
+			];
+			$responses = Promise\unwrap($promises);
```

Comment:
it will throw a ConnectException, if any of requests fails. we can also use settle and wait.
```php
Promise\settle($promises)->wait(); 
```
it will wait for the requests to complete, even if some of them fail.